openmdao/core/driver.py

```diff
@@ -621,36 +633,39 @@ def record_iteration(self):
 
         # Get the data to record
         data = {}
-        if self.options['record_desvars']:
+        if self.recording_options['record_desvars'] or self.recording_options['includes']:
```

Comment:
you might do a preliminary test on `self.recording_options['includes']` alone to short-circuit the following tests (and not have to repeat the 'includes' portion of each test)

Reply:
Perhaps I'm thinking about this incorrectly, but doing a test on includes before desvars/constraints/objectives would be equivalent to and-ing them together rather than or-ing them together, right? With that said, it definitely is unnecessarily verbose; I should store self.recording_options['includes'] in a temporary variable at the very least.

Reply:
Oh nevermind, scratch that. I see what you were saying

Reply:
GJ on understanding my less than clear comment...  it was just a suggestion, no big deal